r/tests/testthat/_snaps/dplyr-join.md

```diff
@@ -0,0 +1,50 @@
+# Error handling
+
+    Code
+      (expect_error(left_join(arrow_table(example_data), arrow_table(example_data),
+      by = "made_up_colname")))
+    Output
+      <error/rlang_error>
+      Join column must be present in data.
+      x `made_up_colname` not present in x.
```

Comment:
The error messages including x/y is a nice touch, though it should be pointed out that does differ from dplyr:

```
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
left_join(mtcars, mtcars, by = "foo")
#> Error: Join columns must be present in data.
#> ✖ Problem with `foo`.
```

Giving the explicit x/y is friendlier though, so I'm happy to keep it.

Reply:
Yes, I thought have explicit x/y error bullets might be better.